packages/cli/src/lib/implementation/core-config.middleware.integration.test.ts

```diff
@@ -35,4 +35,21 @@ describe('coreConfigMiddleware', () => {
       coreConfigMiddleware({ config: 'wrong/path/to/config' }),
     ).rejects.toThrow(/Provided path .* is not valid./);
   });
+
+  it('should load config which relies on provided --tsconfig', async () => {
+    await expect(
+      coreConfigMiddleware({
+        config: join(configDirPath, 'code-pushup.needs-tsconfig.config.ts'),
+        tsconfig: 'tsconfig.base.json',
+      }),
+    ).resolves.toBeTruthy();
+  });
+
+  it('should throw if --tsconfig is missing but needed', async () => {
```

Comment:
But needed for what? Maybe obvious... so not sure if there is more info needed.

Reply:
Adjusted the test description. There's more info in the `code-pushup.needs-tsconfig.config.ts` file.